webapp/src/app.tsx

```diff
@@ -437,6 +437,19 @@ export class ProjectView
         this.forceUpdate(); // we now have editors prepared
     }
 
+    // Add an error guard for the entire application
+    componentDidCatch(error: any, info: any) {
+        core.killLoadingQueue();
```

Comment:
don't do anything if hasError is already set.

Reply:
I'm logging a criticalError event. I believe app insights already logs those anyway but it'd be good to know about the frequency of this event and what component is problematic. 

If react catches a number of exceptions we probably want to log all of them, what's that harm in doing this for all errors? (stop loading is harmless if done more than once, and same with set state) and it's going to all reload in 2 seconds anyway